automation/sheet_checker.py

```diff
@@ -0,0 +1,165 @@
+        ep = ExecutePreprocessor(timeout=600, kernel_name='python3')
+        self.run_notebook(ep, nb, out_notebook_name)
+        self.logger.info("Successfully compressed.")
+        return True
+
+    def run_notebook(self, ep, nb, out_notebook_name):
+        try:
+            ep.preprocess(nb, {'metadata': {'path': '../data_cleaning/'}})
+        except CellExecutionError:
+            msg = 'Error executing the notebook'
+            msg += 'See notebook "%s" for the traceback.' % out_notebook_name
+            self.logger.info(msg)
+            raise
+        finally:
+            with open(out_notebook_name, mode='wt') as f:
+                nbformat.write(nb, f)
+
+
+    def is_sheet_updated(self):
+        sheet_last_updated_ts = dateutil.parser.parse(self.get_sheet_update_ts())
+        last_run_ts = self.fetch_last_run_ts()
+        if not last_run_ts:
+            return False
+        else:
+            job_last_run_ts = dateutil.parser.parse(last_run_ts)
+            job_last_run_ts = job_last_run_ts.replace(tzinfo=pytz.UTC)
+            return sheet_last_updated_ts >= job_last_run_ts
+
+
+    def get_sheet_update_ts(self):
+        gc = pygsheets.authorize(service_file='client_secret.json')
+        gc.enableTeamDriveSupport = True 
+        sheet = gc.open_by_key(self.sheet_key)
+        last_updated_ts = sheet.updated
+        return last_updated_ts
+
+    def update_last_ran_ts(self):
+        timestamp = datetime.strftime(datetime.now(), '%Y-%m-%d %H:%M:%S')
+        buff = BytesIO()
+        buff.write(timestamp.encode('utf-8'))
+        buff.seek(0)
+        self.s3_client.put_object(Body=buff, Bucket='tji-timestamps', Key=self.dataset)
+        self.logger.info("Updated %s timestamp." % self.dataset)
+
+    def fetch_last_run_ts(self):
+        try:
+            timestamp = self.s3_client.get_object(Bucket='tji-timestamps', Key=self.dataset)
+            body = timestamp['Body']
+            last_run_ts = body.read()
+            return last_run_ts
+        except ClientError as e:
+            if e.response['Error']['Code'] == "404":
+                self.logger.info("No timestamp found.")
+            else:
+                self.logger.exception(e)
+                self.logger.info("Something went wrong while fetching timestamp.")
+
+    def set_up_environment(self):
+        dataset = self.dataset.upper()
+        os.environ['CLEAN_%s_S3' % dataset] = 'TRUE'
+        os.environ['COMPRESS_%s_S3' % dataset] = 'TRUE'
+        os.environ['COMPRESS_DATASET'] = dataset
+
+    def clean_up_environment(self):
+        dataset = self.dataset.upper()
+        os.unsetenv('CLEAN_%s_S3' % dataset)
+        os.unsetenv('COMPRESS_%s_S3' % dataset)
+        os.unsetenv('COMPRESS_DATASET')
+
+class CDRChecker(SheetChecker):
+    def __init__(self, *args, **kwargs):
+        super(CDRChecker, self).__init__(*args, **kwargs)
+
+class OISChecker(SheetChecker):
```

Comment:
What's the point of these two subclasses? Why not just instantiate SheetChecker objects directly?

Reply:
I had intended to initialize them with some parameters that were specific to each dataset, but forgot to move the initialization into the subclasses. After a second look, though, I think creating sheet checkers instances from a yaml file makes more sense.